### PR TITLE
 /contest/:id/submission 

### LINKED ISSUE #70: submission controller
/contest/:id/submission

### PR DESCRIPTION
**Pull request by jethoo**
### What this PR does (required):
- Created the controller and added the route to contest , which creates submission object.
- Also added validation if the user has previous submission for the particular contest, which avoids creating new submission object

### Screenshots / Videos (required):
Screen Shot 2021-07-15 at 18 28 59
Screen Shot 2021-07-15 at 20 32 16

### Any information needed to test this feature (required):
- Post what steps to take to be able to test this feature (eg. "sign up as a new user and upload a profile picture from the navbar")

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

File: server/controllers/submission.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      contest: req.params.id,
+      user: req.user.id,
+    });
+
+    if (previousSubmission) {
+      previousSubmission.storeUrl = previousSubmission.storeUrl.concat(
+        body.storeUrl
+      );
+      await previousSubmission.save();
+      res.status(201).json(previousSubmission);
```
Comment (rajivtitus) [inline server/controllers/submission.js]: Please use status code 200 here. Code 201 is for successful resource creation :)

**Review by rajivtitus — APPROVED** [review]
LGTM! Thanks for taking the feedback from our call and working on this :)

File: server/controllers/submission.js (hunk 1 above)
Comment (jeewanbhusal-1) [reply server/controllers/submission.js]: oh yeah. thanks

**Comment by jeewanbhusal-1** [comment]
> LGTM! Thanks for taking the feedback from our call and working on this :)

No problem. Updated the status code with new commit.

**Pull request merged**